src/base.jl

```diff
@@ -1740,6 +1759,10 @@ function handle_component_removal!(sys::System, dyn_injector::DynamicInjection)
     @assert found
 end
 
+function _handle_component_removal_common!(component)
+    clear_units!(component)
```

Comment:
We were not clearing the units in all removal functions. That should be fixed now.